website/server/models/task.js

```diff
@@ -25,6 +26,13 @@ export let TaskSchema = new Schema({
   type: {type: String, enum: tasksTypes, required: true, default: tasksTypes[0]},
   text: {type: String, required: true},
   notes: {type: String, default: ''},
+  shortName: {
+    type: String,
+    validate: [function validateShortName (val) {
+      return Boolean(this.userId) && !validator.isUUID(val) && val.match(SHORT_NAME_VALIDATION_REGEX);
+    },
+    'Task short names can only contain alphanumeric characters, underscores and dashses. They must be owned by a user.'],
```

Comment:
@lemoness @Alys Is this a good error message for setting an invalid value for the short name?

The intention is that 

1) it can only contain letters, numbers, dashes and underscores.
2) it will error if it is a valid uuid (so we don't get conflicts when looking up tasks by id)
3) it will error if the task is not owned by a user (IE, a challenge task)